lib/go-parser/index.ts

```diff
@@ -131,13 +131,22 @@ async function findGoBinaries(
       const first4Bytes = chunk.toString(encoding, 0, 4);
 
       if (first4Bytes === elfHeaderMagic) {
+        // Now that we know it's an ELF file, allocate the buffer
+        buffer = Buffer.alloc(streamSize ?? elfBuildInfoSize);
+
         Buffer.from(chunk).copy(buffer, bytesWritten, 0);
         bytesWritten += chunk.length;
+
         // Listen to next chunks only if it's an ELF executable
         stream.addListener("data", (chunk) => {
-          Buffer.from(chunk).copy(buffer, bytesWritten, 0);
-          bytesWritten += chunk.length;
+          if (buffer && bytesWritten < buffer.length) {
+            Buffer.from(chunk).copy(buffer, bytesWritten, 0);
```

Comment:
should we be adding a fourth parameter here of `buffer.length - bytesWritten` for safety?